src/app/core/models/templates/team-response-template.model.ts

```diff
@@ -4,19 +4,31 @@ import { DuplicateOfSection } from './sections/duplicate-of-section.model';
 import { Section } from './sections/section.model';
 import { Header, Template } from './template.model';
 
+const { everyCharUntil, str, sequenceOf, whitespace } = require('arcsecond');
+
 export const TeamResponseHeaders = {
   teamResponse: new Header("Team's Response", 1),
   duplicateOf: new Header('Duplicate status \\(if any\\):', 2)
 };
 
+const TEAM_RESPONSE_HEADER = "# Team's Response";
+const DUPLICATE_OF_HEADER = '## Duplicate status (if any):';
+
+const TeamResponseParser = sequenceOf([
+  str(TEAM_RESPONSE_HEADER),
+  whitespace,
+  everyCharUntil(str(DUPLICATE_OF_HEADER)),
+  str(DUPLICATE_OF_HEADER)
+]);
+
 export class TeamResponseTemplate extends Template {
   teamResponse: Section;
   duplicateOf: DuplicateOfSection;
   comment: IssueComment;
   parseError: boolean;
 
   constructor(githubComments: GithubComment[]) {
-    super(Object.values(TeamResponseHeaders));
+    super(TeamResponseParser, Object.values(TeamResponseHeaders));
 
     const comment = githubComments.find((githubComment: GithubComment) => this.test(githubComment.body));
     if (comment === undefined) {
```

Comment:
Not completely related to the PR but let's abstract the error handling into Template as well, so that logging would be simpler. Since not all templates make use of this.parseError, we can use a separate attribute Template, (eg this.parseFailure) and set `get parseError() = super.parseFailure` as necessary.

Reply:
For this I think I'll abstract the entire
```typescript
const templateConformingComment = githubIssueComments.find((comment) => this.test(comment.body));
if (templateConformingComment === undefined) {
  this.parseError = true;
  return;
}
```
part into Template since most templates (except Tutor Moderation Issue) has something like this in its constructor. In addition, it should make logging simpler since there's a centralised method to search for matching comments. 

This also requires the GithubComment array as an argument of the constructor of Template. I'll write a separate PR for this